arcgis-ios-sdk-samples/Route and directions/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapStyle: .arcGISNavigation)
+        }
+    }
+    /// The label to display satellites info.
+    @IBOutlet var statusLabel: UILabel!
+    /// The button to start the demo.
+    @IBOutlet var startBarButtonItem: UIBarButtonItem!
+    /// The button to reset pan mode to "recenter".
+    @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
+    /// The button to reset the demo.
+    @IBOutlet var resetBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Instance properties
+    
+    /// An NMEA location data source, to parse NMEA data.
+    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    /// A mock data source to read NMEA sentences from a local file, and generate
+    /// mock NMEA data every fixed amount of time.
+    let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    
+    // MARK: Actions
+    
+    @IBAction func start() {
+        // Set buttons states.
+        startBarButtonItem.isEnabled = false
+        resetBarButtonItem.isEnabled = true
+        // Set NMEA location data source for location display.
+        mapView.locationDisplay.dataSource = nmeaLocationDataSource
+        // Start the data source and location display.
+        mockNMEADataSource.start()
+        mapView.locationDisplay.start()
+        // Recenter the map and set pan mode.
+        recenter()
+    }
+    
+    @IBAction func recenter() {
+        mapView.locationDisplay.autoPanMode = .recenter
+        recenterBarButtonItem.isEnabled = false
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
+            DispatchQueue.main.async { [weak self] in
```

Comment:
`self` is already `weak` in this context.

```suggestion
            DispatchQueue.main.async {
```